components/brave_wallet/browser/cardano/cardano_transaction_serializer.cc

```diff
@@ -235,4 +223,77 @@ bool CardanoTransactionSerializer::ValidateMinValue(
   return min_ada_required && output.amount >= min_ada_required.value();
 }
 
+// static
+bool CardanoTransactionSerializer::AdjustFeeAndOutputsForTx(
+    CardanoTransaction& tx,
+    const cardano_rpc::EpochParameters& epoch_parameters) {
+  CardanoTransaction& result = tx;
+
+  base::CheckedNumeric<uint64_t> total_inputs_amount =
+      result.GetTotalInputsAmount();
+
+  tx.set_fee(0);
+  if (tx.ChangeOutput()) {
+    tx.ChangeOutput()->amount = 0;
```

Comment:
This looks dangerous, if due a mistake result of the function will be used after false is returned, tx change could disappear.
Can we return optional of a new transaction with modified fee instead of modifying existing one?